### PR TITLE
[WIP] [Don't Merge] Utilize frontend-component-header for studio header

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
### todo list
* [x] use helper classes instead of making new ones where possible/appropriate

### CODE REVIEW

File: src/index.scss
Comment (arbrandes) [inline src/index.scss:17]: I think we figured it out, right?

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [inline src/library-authoring/author-library/LibraryAuthoringPage.jsx:270]: Did you remove it as a UX consideration, or just to simplify the refactor temporarily?

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [inline src/library-authoring/author-library/LibraryAuthoringPage.jsx:388]: +1.  As we discussed previously, mobile is not currently a priority for Studio-like MFEs such as this one.

File: src/library-authoring/course-import/CourseImportPage.jsx
Comment (arbrandes) [inline src/library-authoring/course-import/CourseImportPage.jsx:100]: If you look at the console, you'll see React complaining about the missing `key={}`.  I think we need to add it back in.  (Like in the AuthoringPage.)

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [inline src/library-authoring/author-library/LibraryAuthoringPage.jsx:404]: We're missing a `key={}` here: React is throwing a console error because of it.

File: src/library-authoring/course-import/CourseImportTaskListItem.jsx
Comment (arbrandes) [inline src/library-authoring/course-import/CourseImportTaskListItem.jsx:72]: Reminder to remove the reference material :)

File: src/library-authoring/edit-block/LibraryBlockPage.jsx
Comment (arbrandes) [inline src/library-authoring/edit-block/LibraryBlockPage.jsx:223]: Agreed.  I think it's fine to get rid of the redundancy.

File: src/library-authoring/list-libraries/index.scss
Comment (arbrandes) [inline src/library-authoring/list-libraries/index.scss:149]: Reminder to get rid of the saved reference.

File: src/library-authoring/studio-header-wrapper/ContentTitleBlock.jsx
Comment (arbrandes) [inline src/library-authoring/studio-header-wrapper/ContentTitleBlock.jsx:10]: Thanks for sticking with a11y. 👍🏼 

(Disclaimer: I'm not doing a proper accessibility review here.  This is going to be the topic of a separate discussion.  Simply using more Paragon helps a lot, though.)

File: src/vendor/overrides.scss
Comment (arbrandes) [inline src/vendor/overrides.scss:27]: Yeah, we definitely want to move stuff off of `vendor`.  We can leave that for the next issue to tackle: https://github.com/openedx/frontend-app-library-authoring/issues/72

**Review by arbrandes — CHANGES_REQUESTED** [review]
Excellent work, @brian-smith-tcril!  I totally approve of the direction: I find the whole thing to be much more pleasant to navigate.  To be perfectly honest, modulo removal of some commented code and a couple of console issues, I would consider this to be Almost Done ™️.  Congrats!

Btw, I think it's a good idea to consolidate commits somewhat before merging.  I'll leave it to you to pick locus themes.

Another note: there are some feature flag-hidden UI elements to this MFE that I haven't tested, yet, but since I know for a fact the underlying feature I'm thinking of is not actually working at the moment (LTI 1.3 support), I'll leave that to yet another issue to put in the backlog.

**Comment by arbrandes** [comment]
Two things I forgot to note above:

* I tested everything I could and things Just Worked (kudos!)
* We'll also want to fix the linting issues prior to merging, of course

File: src/index.jsx
Comment (brian-smith-tcril) [reply src/index.jsx:41]: my main issue with this implementation is having multiple route switches in the `index.jsx` file here

i did add it in https://github.com/openedx/frontend-app-library-authoring/commit/f9349871fa7dac7bea9a8580d8df3d7bdbb5b19e to clear the state (since `loadingStatus` would stay `loaded` when navigating back to the list page)

i was hoping i'd have some better ideas by now, but nothing is coming to mind

File: src/index.scss
Comment (brian-smith-tcril) [reply src/index.scss:17]: yup! removed in https://github.com/brian-smith-tcril/frontend-app-library-authoring/commit/f04845484162f73a9cbce7fa714cddeb8dd5eb9c

File: src/library-authoring/course-import/CourseImportPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/course-import/CourseImportPage.jsx:100]: fixed by https://github.com/brian-smith-tcril/frontend-app-library-authoring/commit/ff8d45a0a9f637a65b7d59f6b3d72330abbb302c

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx:404]: turns out it was somewhere else (thanks for finding it!) https://github.com/brian-smith-tcril/frontend-app-library-authoring/commit/b56b7d2adf60db4c4e91da77eb8923030f907b33

File: src/library-authoring/edit-block/LibraryBlockPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/edit-block/LibraryBlockPage.jsx]: removed https://github.com/brian-smith-tcril/frontend-app-library-authoring/commit/9edd6f5f3a0936cd35b9b892055da724c537d603

File: src/library-authoring/course-import/CourseImportTaskListItem.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/course-import/CourseImportTaskListItem.jsx]: addressed in https://github.com/openedx/frontend-app-library-authoring/commit/42c82d41cd835ecf990db9a694c02cf19983f998

File: src/library-authoring/list-libraries/index.scss
Comment (brian-smith-tcril) [reply src/library-authoring/list-libraries/index.scss]: addressed in https://github.com/openedx/frontend-app-library-authoring/commit/42c82d41cd835ecf990db9a694c02cf19983f998

File: src/index.jsx
Comment (arbrandes) [reply src/index.jsx:41]: For what it's worth, it seems to me like a legitimate use of `Switch`, and a reasonable place to put it in. 🤷🏼

File: package.json
Comment (arbrandes) [reply package.json:40]: Excellent, thank you!

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx:270]: I'm fine with getting rid of the button, for now.  In the end it won't be a decision to make on our own, but we can always add the button back if somebody complains.

(What will eventually happen, I expect, is that this page will suffer a major UX overhaul.  Whoever develops the changes will likely get handed down a full design document.)

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx:84]: This is one of those things that will have to be brought up in a UX/Design discussion with stakeholders.  For now, I think the best place to keep track of these ideas is in issues in this repository: either all of them in a single one, or tracking them separately (but as subtasks of another).

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (arbrandes) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx:80]: No objections at all.

File: src/library-authoring/library-access/LibraryAccessPage.jsx
Comment (arbrandes) [reply src/library-authoring/library-access/LibraryAccessPage.jsx:123]: Please do!  I know for a fact this page was whipped up with no UX supervision, so we have some leeway.

File: src/index.jsx
Comment (brian-smith-tcril) [reply src/index.jsx:41]: sounds good, i'll remove the todo

File: src/library-authoring/library-access/LibraryAccessPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/library-access/LibraryAccessPage.jsx]: removed in https://github.com/openedx/frontend-app-library-authoring/pull/69/commits/e38d674e12e58c1c89e222120769e7806df0f91b

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx]: made an issue, https://github.com/openedx/frontend-component-header/issues/265, removed the comment https://github.com/openedx/frontend-app-library-authoring/pull/69/commits/fbef590442640ce45f8fd1867d6ba4f3ab675451

File: src/library-authoring/author-library/LibraryAuthoringPage.jsx
Comment (brian-smith-tcril) [reply src/library-authoring/author-library/LibraryAuthoringPage.jsx]: made an issue https://github.com/openedx/frontend-component-header/issues/266, removed the commented line https://github.com/openedx/frontend-app-library-authoring/pull/69/commits/720bc0caa5d4ef0139fee44ee9aaa7938d4e04ef

**Pull request closed without merging**
